mqtt/src/main/java/org/tweetwallfx/mqtt/SystemInfo.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2023 TweetWallFX
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.tweetwallfx.mqtt;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.net.NetworkInterface;
+import java.net.SocketException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class SystemInfo {
+    private static final Logger LOG = LoggerFactory.getLogger(SystemInfo.class);
+
+    public static Map<String, List<String>> info() {
+        try {
+            var infos = new HashMap<String, List<String>>();
+            var interfaces = NetworkInterface.getNetworkInterfaces();
+            while (interfaces.hasMoreElements()) {
+                var networkInterface = interfaces.nextElement();
+                if (!networkInterface.isLoopback()) {
+                    var nicName = networkInterface.getName();
+                    var addresses = networkInterface.getInetAddresses();
+                    while (addresses.hasMoreElements()) {
+                        var address = addresses.nextElement();
+                        if (!address.isAnyLocalAddress() && !address.isLoopbackAddress()) {
+                            infos.computeIfAbsent(nicName, k -> new ArrayList<>()).add(address.getHostAddress());
```

Comment:
that deeeep but due to the SocketException from isLoopback there's not much that can be done about this. switching to streaming operations is out because of that :(